kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,418 @@
+                   'AWS region (or zone) name, like us-west-2 or eu-west-1.'),
+        'cluster': (str, REQUIRED,
+                    'ECS cluster in the region to launch the task into.'),
+        'task_definition':
+            (str, REQUIRED,
+             'String of path to the Task Definition file template. '
+             'Must be a local file path. '
+             'Parameters to be interpolated must be of the form %VAR%.'),
+        'parameters': (dict, {},
+                       'A dictionary of key/value pairs '
+                       'used to fill in the parameters for the template.'),
+        'count': (int, 1, 'How many tasks to run.'),
+        'wait': (bool, True,
+                 'Whether to wait for the tasks to complete.')
+    }
+
+    @gen.coroutine
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision').
+        """
+        # Validate the schema.
+        try:
+            jsonschema.validate(task_definition, TASK_DEFINITION_SCHEMA)
+        except jsonschema.exceptions.ValidationError as e:
+            raise exceptions.InvalidOptions(e)
+
+        family = task_definition['family']
+        if self._dry:
+            self.log.info(
+                'Would register task definition with family {0}'.format(
+                    family))
+            raise gen.Return((family, 1))
+
+        self.log.info('Registering task definition with family {0}'.format(
+            family))
+        response = yield self.thread(
+            self.ecs_conn.register_task_definition, **task_definition)
+
+        # Parse data from the server's response.
+        task_definition = response['taskDefinition']
+        family = task_definition['family']
+        revision = task_definition['revision']
+
+        self.log.info('Task definition {0}:{1} registered'.format(
+            family, revision))
+        raise gen.Return((family, revision))
+
+    @gen.coroutine
+    @dry('Would run task {family}:{revision}')
+    def _run_task(self, family, revision):
+        """Runs a task on ECS given a task definition's family and revision.
+
+        A task can result in multiple running tasks,
+        depending on count and sidekick tasks.
+
+        Args:
+            family: Task family string
+            revision: Task revision string
+
+        Returns:
+            list: task ARNs.
+        """
+        response = yield self.thread(
+            self.ecs_conn.run_task,
+            cluster=self.option('cluster'),
+            taskDefinition='{0}:{1}'.format(family, revision),
+            count=self.option('count'))
+        self._handle_failures(response['failures'])
+        tasks = [t['taskArn'] for t in response['tasks']]
+        raise gen.Return(tasks)
+
+    @gen.coroutine
+    @dry('Would wait for tasks')
+    def _wait_for_tasks(self, tasks):
```

Comment:
Overall I don't love this method.. a few nits. First, it doesn't provide any feedback to the user that it "waited" and is going to check again in 5s. This is where the `repeating_log` logic that Mikhail wrote is pretty useful. Second, you wrap it in a @dry decorator when its not really necessary. If it gets an empty list of tasks (or None), it should just return.